nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -485,6 +505,8 @@ in
         PrintMotd no # handled by pam_motd
 
         AuthorizedKeysFile ${toString cfg.authorizedKeysFiles}
+        AuthorizedKeysCommand ${cfg.authorizedKeysCommand}
+        AuthorizedKeysCommandUser ${cfg.authorizedKeysCommandUser}
```

Comment:
Doesn't this cause issues for people who added `AuthorizedKeysCommand*` at the beginning of sshd.conf? (maybe make it `null` by default and only add those lines if the value is not `null`).

Reply:
@Ma27 I'm not familiar on how to add text to the beginning of `sshd.conf` with NixOS. The only option I'm aware of is `extraConfig`, which goes at the end of the file.

Reply:
``` nix
{ services.openssh.extraConfig = mkBefore ''
  '';
}
```

Or am I misunderstanding you?

Reply:
You understand me correctly. The issue was me! I entirely missed this line:
https://github.com/NixOS/nixpkgs/blob/5529ffdf2fb5ecfab0ad9a5c2085f13790303c40/nixos/modules/services/networking/ssh/sshd.nix#L473

I didn't realize the entire file was pushed into `extraConfig`.

But... `mkBefore` is defined as `mkOrder 500`, so this isn't an issue - the config is merged properly.

On top of that the complexity of actually making the `AuthorizedKeysCommand` work on NixOS makes me think an entry in the release notes might be sufficient (because I assume not many people actually use this feature of `openssh` on NixOS)... but I'm happy to listen to whatever the majority agree to.

Reply:
> But... mkBefore is defined as mkOrder 500, so this isn't an issue - the config is merged properly.

In theory it would be possible to prepend values with `mkOrder -1`, right? ;)
(haven't tested it though)

> On top of that the complexity of actually making the AuthorizedKeysCommand work on NixOS makes me think an entry in the release notes might be sufficient (because I assume not many people actually use this feature of openssh on NixOS).

Seems reasonable to me :+1: